active/tab/tabcontrol.md

```diff
@@ -0,0 +1,580 @@
+| DependencyObject ContainerFromIndex(Int32 index) | Returns the generated container for the specified index. |
+
+### TabViewItem properties and events
+
+| Property | Description |
+|:-------- |:----------- |
+| Content | The main content that appears in the tab. |
+| Header | The content that appears inside the tab itself.  |
+| HeaderTemplate | Template for the header object. |
+| Icon | Icon for the tab. |
+| IsCloseable | Determines if the tab shows a close button. Default is "True". |
+
+| Event | Description |
+|---|---|
+| TabClosing | Fires when a tab's close button is clicked. |
+
+# API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+```
+namespace MU_XC_NAMESPACE
+{
+
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+enum TabViewItemWidthMode
+{
+    Actual = 0,
+    Equal = 1,
+};
+
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+runtimeclass TabViewTabClosingEventArgs
+{
+    Object Item { get; };
+
+    Boolean Cancel { get; set; };
+}
+
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+unsealed runtimeclass TabView : Windows.UI.Xaml.Controls.Control
+{
+    TabView();
+
+    [MUX_DEFAULT_VALUE("winrt::TabViewItemWidthMode::Actual")]
+    [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+    TabViewItemWidthMode TabViewItemWidthMode{ get; set; };
+
+    // TODO: Remove
+    [MUX_DEFAULT_VALUE("true")]
+    Boolean CanCloseTabs{ get; set; };
+
+    [MUX_DEFAULT_VALUE("true")]
+    Boolean CanDragTabs{ get; set; };
+
+    // TODO: Add
+    [MUX_DEFAULT_VALUE("true")]
+    Boolean CanReorderTabs{ get; set; };
+
+    Object TabStripFooterContent{ get; set; };
+    Windows.UI.Xaml.DataTemplate TabStripFooterContentTemplate{ get; set; };
+    
+    Object TabStripHeader{ get; set; };
+    Windows.UI.Xaml.DataTemplate TabStripHeaderTemplate{ get; set; };
+
+    [MUX_DEFAULT_VALUE("true")]
+    Boolean IsAddTabButtonVisible{ get; set; };
+    Windows.UI.Xaml.Input.ICommand AddTabButtonCommand{ get; set; };
+    Object AddTabButtonCommandParameter{ get; set; };
+
+    event Windows.Foundation.TypedEventHandler<TabView, TabViewTabClosingEventArgs> TabClosing;
+
+    // TODO: Update parameter to TabViewAddTabButtonClickEventArgs
+    event Windows.Foundation.TypedEventHandler<TabView, TabViewAddTabButtonClickEventArgs> AddTabButtonClick;
+
+    // TODO: Add
+    event Windows.Foundation.TypedEventHandler<TabView, TabDraggedOutsideEventArgs> TabDraggedOutside;
+
+    // TODO: Update
+    // From ListView
+    [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+    Object TabItemsSource;
+
+    // TODO: Update
+    [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+    Windows.Foundation.Collections.IVector<Object> TabItems{ get; };
+
+    Windows.UI.Xaml.DataTemplate ItemTemplate;
+    Windows.UI.Xaml.Controls.DataTemplateSelector ItemTemplateSelector{ get; set; };
+
+    [MUX_DEFAULT_VALUE("-1")]
+    [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+    Int32 SelectedIndex;
+
+    [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+    Object SelectedItem;
```

Comment:
What happens if `SelectedItem == null` (aka `SelectedIndex == -1`)?

Reply:
Discussion: Behaves like ListView, it's OK to have nothing selected. (No tab is selected and the content area is empty)